### ISSUE TITLE
Usage with raw data

### ISSUE THREAD

**Issue by slhck** [labradon/yuvio#1]
Thanks for this useful library! Can I use this if only have the raw data?

For instance, PyAV returns `frame.planes` as a list, with three entries for YUV 4:2:0, or one entry for UYVY 4:2:2. I want to convert the UYVY representation to YUV 4:2:0 using `yuvio`. How can I construct a `frame` when I don't have the Y/U/V components separated already?

For reference, I'm thinking about using this here:

https://github.com/VQEG/siti-tools/blob/2eb188f1f1a812bb5f96d5a330ee2fe7ad1b486c/siti_tools/file.py#L55-L67

**Comment by labradon** [comment]
Great to hear that the library can be useful to others as well.

Do I understand correctly that you'd like the functionality to
1. separate the individual yuv planes from an interleaved format
2. perform automatic chroma channel sub-/upsampling depending on the source and target subsampling format?

The first functionality is something that I've thought about as well for a while. Hearing that it might be useful for more people than just me could make it worth implementing. I'll give it a go.

The second functionality is something I haven't thought about yet, but definitely sounds useful. Would you like to have a "security" argument like `chroma_resampling_ok` that is `False` by default to prevent accidental resampling?

**Comment by slhck** [comment]
Yes, the first functionality is what I was referring to.

To give you a bit more background: we are getting pixels in packed/interleaved format and need just the Y plane to do some calculations. Originally I thought I could use PyAV' `grey` conversion, which it does through FFmpeg, but I believe it does a weighting of the Y/U/V components. So this is not working either. My fallback was to use `frame.planes[0]`, but this does not work for packed formats like UYVY.

Since our use case is just handling UYVY as the (probably) second most common non-YUV format, resampling hadn't occurred to me. So, automatic sub-/upsampling is a bit more challening, and having just the former would be nice, like:

```python
rawdata = […] # np.ndarray
f = yuvio.frame(rawdata, format = "uyvy422")
f.y # returns just the Y plane
```

**Comment by labradon** [comment]
Great, thanks for the input! I just pushed an updated version and uploaded it to PyPI that is able to do what you are referring to.

It turned out that there was still a problem with interleaved formats and the `YUVFrame` container, I fixed that. Solving your problem then was quite straightforward by allowing to read and write not only to/from file objects but arbitrary `io` streams. By wrapping the in-memory data with `io.BytesIO`, de-interleaving can be performed using `imread`.

```python
import io
import yuvio

rawdata = [...]  # np.ndarray
f = yuvio.imread(io.BytesIO(rawdata), width, height, format="uyvy422")
f.y   # returns just the Y plane
```

The only additional information that is required is the resolution of the Y plane (`width`, `height`).
